src/ledger/LedgerManagerImpl.cpp

```diff
@@ -1555,7 +1556,8 @@ LedgerManagerImpl::prefetchTransactionData(AbstractLedgerTxnParent& ltx,
                                            Config const& config)
 {
     ZoneScoped;
-    if (config.PREFETCH_BATCH_SIZE > 0)
+    if (config.PREFETCH_BATCH_SIZE > 0 &&
+        config.BUCKETLIST_DB_INDEX_PAGE_SIZE_EXPONENT > 0)
```

Comment:
Might be nice to introduce a helper function in Config class `allBucketsInMemory` or something like that, as it's not immediately obvious why it's safe to skip prefetching based on BUCKETLIST_DB_INDEX_PAGE_SIZE_EXPONENT